### ISSUE TITLE
Meross suddenly stopped working

### ISSUE THREAD

**Issue by chowdarygm** [donavanbecker/homebridge-meross#54]
**Describe Your Problem:**
Meross device suddenly stopped working after working for couple of days and showing below error in logs. IP address didn't change and tried rebooting Pi and Homebridge multiple times with no difference. 

**Logs:**
[7/20/2020, 19:55:46] [Garage door] Failed to POST to the Meross Device MSG100 at 192.168.1.16: Error: connect ETIMEDOUT 192.168.1.16:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) {
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '192.168.1.16',
  port: 80
}
(node:1241) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'debug' of undefined
    at /usr/local/lib/node_modules/homebridge-meross/index.js:566:29
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:1241) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 19)

**Comment by dylanfrankcom** [comment]
Maybe @CocoaBob could take a look since they implemented support for the MSG100.

**Comment by CocoaBob** [comment]
I think, ping the device every 2 seconds is too frequent, the the device stops responding... Let me change it to 5 seconds.

**Comment by chowdarygm** [comment]
removed the plugin and re-added it with the merged code and today also getting the same error

[7/23/2020, 11:31:06 AM] [Garage door] Failed to POST to the Meross Device MSG100 at 192.168.1.16: Error: connect ETIMEDOUT 192.168.1.16:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) {
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '192.168.1.16',
  port: 80
}
(node:1375) UnhandledPromiseRejectionWarning: TypeError: this.log is not a function
    at /usr/local/lib/node_modules/homebridge-meross/index.js:558:24
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:1375) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 32)

**Comment by CocoaBob** [comment]
Hi @chowdarygm,

The above error message mainly means the script can't connect to your device at 192.168.1.16 .

Does it work right after reinstalling the plugin? If it doesn't work at all, I'm afraid your Meross device isn't connected to the internet now, or its IP address has been changed (re-assigned by your router)

Please try something like this:
1. Check if you can control the opener in the official Meross app
2. Check if you can ping the local address 192.168.1.16 at home, from your computer

If anything one of them is failed, then I guess you may need to reboot your Meross device (just replug the power), or re-configure its IP address.

**Comment by chowdarygm** [comment]
> Hi @chowdarygm,
> 
> The above error message mainly means the script can't connect to your device at 192.168.1.16 .
> 
> Does it work right after reinstalling the plugin? If it doesn't work at all, I'm afraid your Meross device isn't connected to the internet now, or its IP address has been changed (re-assigned by your router)
> 
> Please try something like this:
> 
> 1. Check if you can control your garage opener in the official Meross app
> 2. Check if you can ping the local address 192.168.1.16 at home, from your computer (just type `ping 192.168.1.16` in Windows command line or Mac Terminal).
> 
> If any one of them is failed, then I guess you may need to reboot your Meross device (just replug the power), or re-configure its IP address.

I think I found the culprit but that seems to be  not the best solution for me. Meross is on my guest network and Homebridge is on the main router. usually I don't let devices to see each other between Guest and main network but disabling that feature and letting them see each other seems to fix the issue. I had to reset the router as other devices also were getting wonky, enabled and set the new IP in settings and restarted. For the last 10 minutes, it has been little stable and working fine but it may be an issue as I would like to keep all IoT devices separate to my actual network except Pi to have free access to it.

[7/23/2020, 1:28:45 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:45 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:46 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:47 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:50 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:50 PM] [Garage door] Failed to POST to the Meross Device MSG100 at 192.168.1.18: Error: connect ETIMEDOUT 192.168.1.18:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) {
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '192.168.1.18',
  port: 80
}
(node:7153) UnhandledPromiseRejectionWarning: TypeError: this.log is not a function
    at /usr/local/lib/node_modules/homebridge-meross/index.js:558:24
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:7153) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 62)
[7/23/2020, 1:28:50 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:52 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:55 PM] [Garage door] Current state CLOSED
[7/23/2020, 1:28:55 PM] [Garage door] Current state CLOSED
